src/reader.rs

```diff
@@ -626,18 +638,18 @@ impl<R: BufRead> Reader<R> {
                 self.ns_resolver.push(&e, namespace_buffer);
```

Comment:
The change makes sense, but just checking - the user could also avoid this whole scenario by enabling the expansion of Empty tags into Start and End tags, right?  This is a workaround for this one specific scenario when that feature isn't enabled.